### ISSUE TITLE
Potential issue in rgba to rgb conversion

### ISSUE THREAD

**Issue by sidd-27** [kornia/kornia#3311]
### Describe the bug

    a_one = torch.tensor(1.0) - a
    r_new: Tensor = a_one * r + a * r
    g_new: Tensor = a_one * g + a * g
    b_new: Tensor = a_one * b + a * b

this is the current code in the rgba to rgb conversion, here if you look closely all terms with a cancel out leaving just r g and b, this seems wrong, and is diffrent from the algorithm mentioned here https://stackoverflow.com/questions/2049230/convert-rgba-color-to-rgb

so if this is indeed the intended version, i am open to making a pr by replacing the math with a slice operation, otherwise ill implement the algorithm in the stackoverflow link

### Reproduction steps

```bash
1.
2.
3.
...
```

### Expected behavior

it being not that ig

### Environment

```shell
wget https://raw.githubusercontent.com/pytorch/pytorch/main/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py

- PyTorch Version (e.g., 1.0):
- OS (e.g., Linux):
- How you installed PyTorch (`conda`, `pip`, source):
- Build command you used (if compiling from source):
- Python version:
- CUDA/cuDNN version:
- GPU models and configuration:
- Any other relevant information:
```

**Comment by edgarriba** [comment]
please go ahead with the update

**Comment by rehman-batt** [comment]
hi, 
can i take this up and fix it, if no one else is working on it?

**Comment by sidd-27** [comment]
sure @rehman-batt  go ahead

**Comment by rehman-batt** [comment]
the algorithm given at the stack overflow link assumes that we've a background as well as a source image, however we're using 
the same image as both background and source image in our algorithm. Also it talks about normalizing the source image as well.

I think what we need to take from that algorithm is only the precedence that is set by the (). or is there anything else that we need to take as well?
